### PR TITLE
doc: update DO from Massimo

### PR DESCRIPTION
**Pull request by vruge**
### Related Issues
#11739

### CODE REVIEW

File: doc/UsersGuide/source/OM_DOWithAnnotations.rst [hunk 1]
```diff
@@ -283,18 +323,14 @@ The code is very simple and it is as follows:
    end BangBang2021;
 
 The constraint on power is especially worth considering. Above, we
-stated that path constraints are
-:math:`0 \leq g(\mathbf{x}(t),\mathbf{u}(t))` here we have box limits on
-pow, which is indeed a function of state variables as follows:
-
-.. math:: - 30 \leq fv \leq 30\  = > \  - 30 \leq mav \leq 30
+stated that path constraints can be
+:math:`g_min \leq g(\mathbf{x}(t),\mathbf{u}(t), t) \leg g_max`. Here we have box limits on
```
Comment (phannebohm) [inline doc/UsersGuide/source/OM_DOWithAnnotations.rst:327]: I think this was the reason it didn't build.
```suggestion
:math:`g_min \leq g(\mathbf{x}(t),\mathbf{u}(t), t) \leq g_max`. Here we have box limits on
```

File: doc/UsersGuide/source/OM_DOWithAnnotations.rst [hunk 2]
```diff
@@ -283,18 +323,14 @@ The code is very simple and it is as follows:
...
+`pow`, which are expressed as limits on `g(v,f) = f*v = pow`
 
-And *a* and *v* are two state variables. So, these are two box
-constraints of the type
-:math:`0 \leq \ g\left( \mathbf{x}(t),\mathbf{u}(t) \right)\ ` as
-follows:
+.. math:: - 30 \leq v*f = f*v \leq 30
 
-:math:`mav = m\text{\ x}_{1}\text{\ x}_{2} \geq - 30\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ mav = m\text{\ x}_{1}\text{\ x}_{2} \leq \ 30`.
+This usage of constrains allows implementing variable (non-boxed) constraints, which are not allowed
```
Comment (phannebohm) [inline doc/UsersGuide/source/OM_DOWithAnnotations.rst:332]: Typo?
```suggestion
This usage of constraints allows implementing variable (non-boxed) constraints, which are not allowed
```

File: doc/UsersGuide/source/OM_DOWithAnnotations.rst [hunk 3]
```diff
@@ -234,6 +238,42 @@ Here we just give two examples:
    negligible loss of precision, substituting :math:`abs(x)` with :math:`sqrt(x^2+\varepsilon)`,
    where eps is very low in comparison with the values :math:`x` usually assumes
    during the simulation.
+- Carefully consider having in the code asserts that cause simulation to stop. If for instance we have
```
Comment (phannebohm) [inline doc/UsersGuide/source/OM_DOWithAnnotations.rst:241]: newline and double space?
```suggestion

-  Carefully consider having in the code asserts that cause simulation to stop. If for instance we have
```

File: doc/UsersGuide/source/OM_DOWithAnnotations.rst [hunk 4]
```diff
@@ -234,6 +238,42 @@ Here we just give two examples:
...
+   an assert that stops simulation when a variable gets outside its limiting value, it may happen that
+   during the optimisation cycle the limit is hit and simulation is stopped, which may not desirable.
```
Comment (phannebohm) [inline doc/UsersGuide/source/OM_DOWithAnnotations.rst:243]:
```suggestion
   during the optimisation cycle the limit is hit and simulation is stopped, which may not be desirable.
```

**Review by phannebohm — APPROVED** [review]
Thanks for updating the docs! I think I found a few typos.

**Comment by vruge** [comment]
@phannebohm thanks :)

**Pull request merged**